scripts/make-helpers.sh

```
@@ -17,27 +19,133 @@ hint() {
+
 # run_cmd "Name" <command...>
-# Respects D3K_LOG_DRY_RUN=1 for safe testing. Emits intent, command, timing, exit code.
+# - Emits human-readable intent with timing/exit
+# - Captures stdout/stderr per command, consolidates into combined.log
+# Env knobs:
+#   D3K_LOG_DRY_RUN=1  -> do not execute, still log intent
+#   D3K_LOG_DIR=path   -> where to write logs (default .make-logs)
 run_cmd() {
   local name="$1"; shift
   local cmd=("$@")
   local t0=$(date +%s)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)